src/modal/modal.spec.ts

```diff
@@ -283,6 +285,51 @@ describe('ngb-modal', () => {
       });
     });
 
+    describe('stacked  modals', () => {
+
+      it('should not remove "modal-open" class on body when closed modal is not last', async(() => {
+           const modalRef1 = fixture.componentInstance.open('foo');
+           const modalRef2 = fixture.componentInstance.open('bar');
+           fixture.detectChanges();
+           expect(fixture.nativeElement).toHaveModal();
+           expect(document.body).toHaveCssClass('modal-open');
+
+           modalRef1.close('foo result');
+           fixture.detectChanges();
+           fixture.whenStable().then(() => {
+             expect(fixture.nativeElement).toHaveModal();
+             expect(document.body).toHaveCssClass('modal-open');
+
+             modalRef2.close('bar result');
+             fixture.detectChanges();
+             fixture.whenStable().then(() => {
+               expect(fixture.nativeElement).not.toHaveModal();
+               expect(document.body).not.toHaveCssClass('modal-open');
+             });
+           });
+         }));
+
+      it('should dismiss modals on ESC in correct order', () => {
```

Comment:
<kbd>ESC</kbd> didn't work because `ngbFocusTrap` was preventing focus to go out of the first modal window.

I fixed it by moving `ngbFocusTrap` to the `NgbModalStack` service, but I'm not sure how to test it. The test seems right, but it still passes, no matter where I call `ngbFocusTrap`.

Reply:
Your changes look good to me. No matter where we call the `ngbFocusTrap`, test does not need to be changed I think.